cpp/src/structs/utilities.cpp

```diff
@@ -98,11 +98,20 @@ struct table_flattener {
   std::vector<null_order> flat_null_precedence;
   column_nullability nullability;
 
```

Comment:
No newline here. These go with the other struct elements.
```suggestion
```

Reply:
Actually the member variables are already grouped:
```
table_view input;
  // reference variables
  std::vector<order> const& column_order;
  std::vector<null_order> const& null_precedence;
  // output
  std::vector<std::unique_ptr<column>> validity_as_column;
  temporary_nullable_data nullable_data;
  std::vector<column_view> flat_columns;
  std::vector<order> flat_column_order;
  std::vector<null_order> flat_null_precedence;
  column_nullability nullability;
```

So if removing that empty line, `stream` and `mr` will be grouped with `// output` so I would prefer to keep it here.

Reply:
Sorry, I didn’t read closely enough to realize those were outputs. What I should have said is that these should be grouped with the other constructor arguments, in initialization order. Let’s make that change: put stream and mr before the outputs and after the other constructor arguments.

Reply:
That sound... a bit awkward to me, as `stream` and `mr` typically stay last in the parameter list :|

Reply:
How about this?
```suggestion
// stream and memory resource
```

Reply:
My point is that `flat_columns` and similar are not constructor arguments. Group stream and mr with the other constructor arguments here.

Reply:
Or sorry I misunderstood that. Update pushed.